api/app/models/notification_preference.rb

```diff
@@ -0,0 +1,41 @@
+class NotificationPreference < ApplicationRecord
+  # Constants
+  BOOLEAN_FREQUENCIES = %w(never always).freeze
```

Comment:
These strings should be captured in constants. You reference them elsewhere in code, and I'd prefer to see those replaced with NotificationPreference::NEVER, for example.

Reply:
I would argue this and `DIGEST_FREQUENCIES` should be a classy_enum, not just a constant. It's the direction we want to move to, and these are brand new.

Reply:
Agreed, classy_enum is better

Reply:
The values are, I'm not referencing the enum there.  Is that what you meant or are you saying the actual constants `BOOLEAN_FREQUENCIES` and `DIGEST_FREQUENCIES` should be enums too?

Reply:
The logic for what is valid for digest or boolean should be contained in the classy_enum (that's why they are classy!), and the validator should be reworked.

Reply:
Oh, ok, I think I get it.  Thanks.

Reply:
✔